app/invitations/services/external_services.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+
+from fastapi import Depends
+
+from app import settings
+from app.core.exceptions import OptScaleAPIResponseError
+from app.optscale_api.invitation_api import OptScaleInvitationAPI
+from app.optscale_api.orgs_api import OptScaleOrgAPI
+from app.optscale_api.users_api import OptScaleUserAPI
+
+logger = logging.getLogger(__name__)
+
+
+async def register_invited_user_on_optscale(
+    email: str, display_name: str, password: str
+) -> dict[str, str] | Exception:
+    """
+    Registers invited users to OptScale, without
+    verification.
+
+    :param email: The email of the given user to register
+    :param display_name: The display name of the user
+    :param password: The password of the user
+    :return: dict[str, str] : User information.
+    :raises: OptScaleAPIResponseError if any error occurs
+        contacting the OptScale APIs
+    """
+    user_api = OptScaleUserAPI()
+    try:
```

Comment:
I miss the validation that there's an invitation for such email.

Reply:
Fixed